### ISSUE TITLE
Unexpected route id inconsistency

### ISSUE THREAD

**Issue by juanbercoff** [remix-run/remix#9591]
### Reproduction

https://stackblitz.com/edit/remix-run-remix-5qqadr?file=vite.config.ts,app%2Froutes%2Fdashboard%2Froute.tsx,app%2Froutes%2Fdashboard%2Fchild%2Froute.tsx

### System Info

```ts
import { vitePlugin as remix } from '@remix-run/dev';
import { defineRoutes } from '@remix-run/dev/dist/config/routes';
import { defineConfig } from 'vite';
import tsconfigPaths from 'vite-tsconfig-paths';

export default defineConfig({
  plugins: [
    remix({
      future: {
        v3_fetcherPersist: true,
        v3_relativeSplatPath: true,
        v3_throwAbortReason: true,
      },
      routes(defineRoutes) {
        return defineRoutes((route) => {
          route('dashboard', 'routes/dashboard/route.tsx', () => {
            route('child', 'routes/dashboard/child/route.tsx');
          });
        });
      },
    }),

    tsconfigPaths(),
  ],
});

```

### Used Package Manager

npm

### Expected Behavior

The id of the parent route should be the same regardless the route being rendered.

### Actual Behavior

When rendering the `dashboard` path, the id of the route is `routes/dashboard`, but when rendering the child route,  `dashboard/child` the id of the parent route is `routes/dashboard/route`.

Should the id always be the same relative to app?

**Comment by kiliman** [comment]
The route ID is the filename relative to app folder without the file extension.

**Comment by akomm** [comment]
@kiliman I've created a reproduction that is a bit easier to use:

Remix DOC on ID: https://remix.run/docs/en/main/hooks/use-route-loader-data
Blitzstack: https://stackblitz.com/edit/remix-run-remix-mmhd6w

Maybe remix did there some BC break within non-major version, but as I've mentioned in react-flat-routes before and you verified, the ID is the path of the file, not the pathname (typically including the filename itself). The example in the docs just provides file a file route example, not using folders.

But it makes sense, that the same route, expressed differently, should have the same ID, right?

So `app/routes/teams.tsx` and `app/routes/teams/route.tsx` should have the same id: `routes/teams`, not `routes/teams` and `routes/teams/route`.

**Comment by juanbercoff** [comment]
This is the config with the explicit ids.
![image](https://github.com/remix-run/remix/assets/15874591/d2612bff-7276-4f37-aba5-1abcac971882)

When you navigate to `dashboard/child`, the id is different that when navigating to `dashboard`.
![image](https://github.com/remix-run/remix/assets/15874591/6ed1fcf7-30f0-4c42-8ca3-8bd44b164a43)

**Comment by akomm** [comment]
This is because of this result (note the dashboard is present twice):
![grafik](https://github.com/remix-run/remix/assets/2409779/8fa30232-deac-430f-bb67-07b601bdb7d1)

I think the reason for this is, that in defineRoutes you don't overwrite everything starting from "root" and your dashboard/route.tsx is compatible with the default remix routing. So what happens is remix picks up the remix compatible dashboard/route.tsx and then adds your routing definitions on top.

**Comment by juanbercoff** [comment]
Yep, it looks like it. I think there should be a note on the docs to replace the whole tree to avoid this kind of issues,

**Comment by brophdawg11** [comment]
We should warn about this - the solution arrived at above is accurate - if you want to manually define child routes of an _existing_ file-based parent route, you will need to specify the parent route with the same ID that Remix uses on the file-system.

But our code should be smart enough to warn/error if a dup route is found so you don't end up with 2 `path="dashboard"` routes in the tree.  

Leaving this open so we can take a look.  The check would likely make sense to do [here](https://github.com/remix-run/remix/blob/main/packages/remix-dev/config.ts#L563) if anyone wants to take a stab.  We should be able to detect if we're about to add a config-based route that has the same `parentId`/`path`/`index` values  as an existing route.

**Comment by akomm** [comment]
Do you think it might make sense to have an option to opt-out of remix default routing, so if someone wants to implement a custom solution there is no chance of accidentally leak remix logic into own implementation? Especially when there are changes in the future major releases. The error message is nice, but only triggers when the case is tested. When a future change adds new potential cases, someone would have to think it all through to figure out what can happen. With this is disabled for a custom routing, only evtl. API changes (hopefully typed) need to be addressed.

**Comment by sergiodxa** [comment]
You can use `ignoredRouteFiles: ['**/*']` to ignore every file inside app/routes and then use the routes function to define your routes by hand.
